### PR TITLE
Unit and integration test for DTMF reception on SIP client and timing

### PR DESCRIPTION
**Pull request by biglittlebigben**

### CODE REVIEW

**Comment by biglittlebigben** [comment]
I haven't been able to make sure integration tests still pass as they don't seem to run on OSX (media time out)

**Review by davidzhao — APPROVED** [review]
lgtm!

File: pkg/siptest/client.go [hunk 1]
```diff
@@ -503,12 +524,32 @@ func (c *Client) WaitSignals(ctx context.Context, vals []int, w io.WriteCloser)
...
+
+		select {
+		case <-ctx.Done():
+			close(pkts)
+			close(done)
+			return ctx.Err()
+		case pkts <- pkt:
 		}
+
+		pkts <- pkt
```
Comment (dennwc) [inline pkg/siptest/client.go]: Looks like it's sending `pkt` twice: once here and once in the `select` above.

File: test/integration/sip_test.go [hunk 2]
```diff
@@ -223,7 +224,12 @@ func runClientWithCodec(t testing.TB, conf *NumberConfig, id string, number stri
 		OnMediaTimeout: func() {
 			t.Fatal("media timeout")
 		},
-	})
+	}
+	if onDTMF != nil {
+		cconf.OnDTMF = onDTMF
```
Comment (dennwc) [inline test/integration/sip_test.go]: Could be just set in the struct init above

**Review by dennwc — APPROVED** [review]

**Pull request merged**